editor/src/components/canvas/canvas-utils.ts

```diff
@@ -810,23 +811,42 @@ function updateFrameValueForProp(
 ): ValueAtPath | null {
   if (delta !== 0) {
     const existingProp = frameProps[framePoint]
-    const pinIsPercentage = existingProp == null ? false : isPercentPin(existingProp)
-    let valueToUse: string | number
-    if (existingProp != null && pinIsPercentage) {
-      const percentValue = numberPartOfPin(existingProp)
-      if (parentFrame != null) {
-        const referenceSize = isHorizontalPoint(framePoint) ? parentFrame.width : parentFrame.height
-        const deltaAsPercentValue = (delta / referenceSize) * 100
-        valueToUse = `${percentValue + deltaAsPercentValue}%`
-      } else {
-        valueToUse = `${percentValue + delta}%`
+    if (existingProp == null) {
+      return {
+        path: createLayoutPropertyPath(pinnedPropForFramePoint(framePoint)),
+        value: jsxAttributeValue(delta, emptyComments),
       }
-    } else {
-      valueToUse = existingProp == null ? delta : Number(existingProp) + delta
     }
-    return {
-      path: createLayoutPropertyPath(pinnedPropForFramePoint(framePoint)),
-      value: jsxAttributeValue(valueToUse, emptyComments),
+    const parsedProp = foldEither(
+      (_) => null,
+      (r) => r,
+      parseCSSLengthPercent(existingProp),
+    )
+    if (parsedProp != null) {
+      const pinIsPercentage = parsedProp.unit === '%'
+      const pinIsUnitlessOrPx = parsedProp.unit == null || parsedProp.unit === 'px'
+      let valueToUse: string | number
```

Comment:
this valueToUse could be moved into the `if (pinIsPercentage)` body to limit scope

Reply:
fixed